control/control.go

```diff
@@ -911,8 +1003,43 @@ func (p *pluginControl) CollectMetrics(metricTypes []core.Metric, deadline time.
 
 	// For each available plugin call available plugin using RPC client and wait for response (goroutines)
 	for pluginKey, pmt := range pluginToMetricMap {
+
+		// if plugin does not have a pool, try to make new subscription
+		if !p.pluginRunner.AvailablePlugins().hasPool(pluginKey) {
+			// this happens only when the queried metric in task manifest matches with metrics
```

Comment:
Dealing with plugins being loaded that may match a query by loaded event handler has been implemented and it is in code review (PR https://github.com/intelsdi-x/snap/pull/997), so when that will be accepted/merged, I return to work on that (rebase code and check/fix issues which @candysmurf  reported)